### PR TITLE
Ensuring that we never call kill with empty arguments in local-cluster-up

### LINKED ISSUE #8227: local-up-cluster.sh fails to kill processes it launches
I'm seeing this reliably in master:

```
^CCleaning up...
kill: not enough arguments
kill: not enough arguments
kill: not enough arguments
kill: not enough arguments
kill: not enough arguments
```

### PR DESCRIPTION
**Pull request by nak3**
This PR will fix [issue#8227](https://github.com/GoogleCloudPlatform/kubernetes/issues/8227)

@nikhiljindal Although I checked several environment with root/non-root user, could you please check your environment?

### CODE REVIEW

**Comment by pmorie** [comment]
I'm going to test this locally.

**Comment by pmorie** [comment]
WFM

**Comment by pmorie** [comment]
Ugh, gh fail

**Comment by pmorie** [comment]
I'm cool with this being merged after @nikhiljindal checks it out and tests it.  No point waiting for travis since CI does not exercise this.

**Comment by nikhiljindal** [comment]
Just tried it doesnt work for me. Still the same error.
Trying to see why it fails

**Comment by nikhiljindal** [comment]
ok. This works fine when the API server comes up healthy and then I kill it after my work is done using Ctrl+C.

However, when API server fails to come up (due to bugs such as https://github.com/GoogleCloudPlatform/kubernetes/pull/8174), then 
{APISERVER_PID} is not empty but since the server crashed already,
$(ps -eh --ppid=${APISERVER_PID} --pid=${APISERVER_PID} doesnt return anything.
Hence we pass an empty string to kill and hence I get the following output:

```
Cleaning up...

Usage:
 kill [options] <pid> [...]

Options:
 <pid> [...]            send signal to every <pid> listed
 -<signal>, -s, --signal <signal>
                        specify the <signal> to be sent
 -l, --list=[<signal>]  list all signal names, or convert one to a name
 -L, --table            list all signal names in a nice table

 -h, --help     display this help and exit
 -V, --version  output version information and exit

For more details see kill(1).
```

We can fix it by first checking if the server is still running, before trying to kill it.
ie. call kill only if ``` $(ps -eh --ppid=${APISERVER_PID} --pid=${APISERVER_PID} ``` is non-empty

**Comment by nikhiljindal** [comment]
And I will suggest updating the PR title so that its self explanatory. You can say something like "Fixing the issue where kill was passed an empty string in local-cluster-up" or "Ensuring that we never call kill with empty arguments in local-cluster-up"

**Comment by nak3** [comment]
OK, I will do it. Thank you for your all comments.

**Comment by nak3** [comment]
@nikhiljindal I pushed the updated PR. 
Also I fixed all indent size in the cleanup(), since it is good timing to do it.

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -108,23 +108,30 @@ cleanup_dockerized_kubelet()
...
-
-    [[ -n "${PROXY_PID-}" ]] && sudo kill $(ps --ppid=${PROXY_PID} --pid=${PROXY_PID} | cut -d' ' -f1)
-    [[ -n "${SCHEDULER_PID-}" ]] && sudo kill $(ps --ppid=${SCHEDULER_PID} --pid=${SCHEDULER_PID} | cut -d' ' -f1)
-
-    [[ -n "${ETCD_PID-}" ]] && kube::etcd::stop
-    [[ -n "${ETCD_DIR-}" ]] && kube::etcd::clean_etcd_dir
-
-    exit 0
+  echo "Cleaning up..."
+  [[ -n "${APISERVER_PID-}" ]] && APISERVER_PIDS=$(ps -eh --ppid=${APISERVER_PID} --pid=${APISERVER_PID} |awk '{print $1}'|xargs)
```
Comment (nikhiljindal) [inline hack/local-up-cluster.sh]: I will suggest adding a comment here saying something like: "Check if the API server is still running"

**Comment by nikhiljindal** [comment]
Thanks, LGTM modulo a minor nit.

**Comment by nak3** [comment]
Thank you @nikhiljindal I did it

**Comment by nikhiljindal** [comment]
Thanks. Will merge on green.

**Pull request merged**